crates/autopilot/src/database/quotes.rs

```diff
@@ -16,6 +19,29 @@ impl Postgres {
         database::quotes::remove_expired_quotes(&mut ex, max_expiry).await?;
         Ok(())
     }
+
+    pub async fn read_quotes(&self, auction: &Auction) -> Result<HashMap<OrderUid, Quote>> {
+        let mut quote_tasks = Vec::new();
+        for order in &auction.orders {
+            let mut ex = self.pool.acquire().await?;
+            let order_uid = ByteArray(order.metadata.uid.0);
+            let quote_task = tokio::spawn(async move {
+                database::orders::read_quote(&mut ex, &order_uid)
+                    .await
+                    .ok()?
+            });
+            quote_tasks.push((order.metadata.uid, quote_task));
+        }
+        let mut quotes_map = HashMap::new();
+        for (order_uid, quote_task) in quote_tasks {
+            let quote = quote_task
+                .await?
+                .ok_or(anyhow::anyhow!("failed to parse quote"))?;
```

Comment:
For turning an option or some other error into an anyhow error you can also use `.context("failed to parse quote")`

Reply:
updated